File: Leetcode/9. Palindrome Number/js/palindrome.js
"use strict";

const Palindrome = (x) => {
  if (x < 0) return false;
  let t = x;
  let r = 0;
  while (x) {
    r = r * 10 + (x % 10);
    x = Math.trunc(x / 10);
  }
  // if (r > INT_MAX || r < INT_MIN)
  //     return 0;
  return r == t;
};

console.log(Palindrome(-121));

File: Leetcode/9. Palindrome Number/palindrome.cpp
#include <iostream>
#include <bits/stdc++.h>

using namespace std;

bool isPalindrome(int x)
{
    int t = x;
    long r = 0;
    while (x)
    {
        r = (r * 10) + x % 10;
        x = x / 10;
    }
    if (r > INT_MAX || r < INT_MIN)
        return 0;
    return int(r) == t;
}

int main()
{

    int n;
    cin >> n;

    cout << isPalindrome(n);

    return 0;
}